### ISSUE TITLE
Provision a new `h` user when LMS instances launched

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/product-backlog#712]
This is a rather sizable integrated feature, so if it needs to be broken into multiple cards, we can do that. Depends on https://github.com/hypothesis/product-backlog/issues/700 and #711 

When an embedded Hypothesis LMS app is launched, a new Hypothesis user should be created for the LMS user at hand.

This is done when:

* [ ] A mapping and logic has been created to translate incoming LTI data into an appropriate payload for the `POST users/` endpoint on `h` (i.e. which LTI data fields should be used to generate a username, display name, identity/provider details, etc.)
* [ ] The relevant view in the LMS app sends a request to the create-user endpoint
* [ ] The LMS app can handle user-create failures and distinguish between them appropriately

Out of scope: Authentication—this card is related to _creating_ the user, not _authenticating_ as that user.

**Comment by dwhly** [comment]
I have a dumb question.  How was this done before, for eLife, if we're just building this capability now?

**Comment by robertknight** [comment]
This card is about the lms app creating a user account in h when it needs to. The equivalent logic for eLife is handled by code in eLife's server.

The lms app needs to do one additional thing which did not apply for eLife, which is storing additional information about the original source of the user's identity: the LMS instance and the unique ID of the user within that instance.

**Comment by seanh** [comment]
I guess there may be different LTI launches involved here. For example our app is launched during assignment or module item creation (at this point the user will be the teacher). The app is also launched during assignment launch, at which point the user may be the teacher or a student. I don't think we receive an LTI launch during the app installation process (e.g. when a teacher installs our app into a course).

Anyway, so the user create code may need to be run at multiple different times.

**Comment by seanh** [comment]
I've broken out some other cards that need to be done before this card can be done:

* <del>[Configure the LMS application as an `auth_client` to `h`](https://github.com/hypothesis/product-backlog/issues/711)</del> done
* <del>[Implement method to enable/disable LMS provisioning](https://github.com/hypothesis/product-backlog/issues/699)</del> done
* <del>[Generate h identities from LMS data (must be deterministic)](https://github.com/hypothesis/product-backlog/issues/736)</del> done
* <del>[Generate valid h usernames from LTI data (must be deterministic)](https://github.com/hypothesis/product-backlog/issues/737)</del> done
* <del>[Generate valid h display names from LTI data](https://github.com/hypothesis/product-backlog/issues/738)</del> done

Can be done after this card:
* <del>[LMS app should handle error states during user provisioning](https://github.com/hypothesis/product-backlog/issues/735)</del> done
* [Implement a faster, easier and more effective way to test LTI launches](https://github.com/hypothesis/product-backlog/issues/739)
